metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -267,7 +279,11 @@ final class RenameProvider(
     }
   }
 
-  private def canRenameSymbol(symbol: String, newName: Option[String]) = {
+  private def canRenameSymbol(
+      symbol: String,
```

Comment:
I think you try to bind symbol with AbsolutePath and propagate them together through multiple functions, maybe it is worth making case class containing those 2 things and just pass one thing. Also it will be easier to see this path is related to this symbol

Reply:
Okay actually I would go with symbol <-> List[AbsolutePath] as it looks like there can be multiple definitions of a symbol. This would fix a problem with `<-` above

Reply:
And then you can decide in `symbolIsLocal` if given symbol with all definition locations is local or not. 
Previously we did `.exists` which means at least one location must be in workspace, but maybe `.all` would make more sense here?
I don't know many multi def locations examples, is it only class/trait + object scenario?

Reply:
I don't think we have multiple definitions anywhere currently really. And even if we get one, I can't imagine one being inside workspace and the other outside.

I can rework it a bit, but I don't think we need to add a case class just for 2 methods.

Reply:
Moved `isWorkspaceSymbol` up, so that we don't propagate the path now.